constructor/briefcase.py

```diff
@@ -0,0 +1,153 @@
+"""
+Logic to build installers using Briefcase.
+"""
+
+import logging
+import re
+import shutil
+import sysconfig
+import tempfile
+from pathlib import Path
+from subprocess import run
+
+import tomli_w
+
+from . import preconda
+from .utils import DEFAULT_REVERSE_DOMAIN_ID, copy_conda_exe, filename_dist
+
+BRIEFCASE_DIR = Path(__file__).parent / "briefcase"
+EXTERNAL_PACKAGE_PATH = "external"
+
+logger = logging.getLogger(__name__)
+
+
+def get_name_version(info):
+    name = info["name"]
+    if not name:
+        raise ValueError("Name is empty")
+
+    # Briefcase requires version numbers to be in the canonical Python format, and some
+    # installer types use the version to distinguish between upgrades, downgrades and
+    # reinstalls. So try to produce a consistent ordering by extracting the last valid
+    # version from the Constructor version string.
+    #
+    # Hyphens aren't allowed in this format, but for compatibility with Miniconda's
+    # version format, we treat them as dots.
+    matches = list(
+        re.finditer(
+            r"(\d+!)?\d+(\.\d+)*((a|b|rc)\d+)?(\.post\d+)?(\.dev\d+)?",
+            info["version"].lower().replace("-", "."),
+        )
+    )
+    if not matches:
+        raise ValueError(
+            f"Version {info['version']!r} contains no valid version numbers: see "
+            f"https://packaging.python.org/en/latest/specifications/version-specifiers/"
+        )
+    match = matches[-1]
+    version = match.group()
+
+    # Treat anything else in the version string as part of the name.
+    start, end = match.span()
+    strip_chars = " .-_"
+    before = info["version"][:start].strip(strip_chars)
+    after = info["version"][end:].strip(strip_chars)
+    name = " ".join(s for s in [name, before, after] if s)
```

Comment:
So, for something like `Miniconda3-py310_25.9.1-3-Windows-x86_64.msi`, we have the following?

```
name = Miniconda3 py310
version = 25.9.1.3
```